pkg/controller/template/render.go

```diff
@@ -375,29 +376,32 @@ func skipMissing(key string) (interface{}, error) {
 }
 
 func cloudProvider(cfg RenderConfig) (interface{}, error) {
-	if cfg.Infra.Status.PlatformStatus != nil {
-		if cfg.FeatureGateAccess == nil {
-			panic("FeatureGateAccess is nil")
-		}
+	if cfg.Infra.Status.PlatformStatus == nil {
+		return "", nil
+	}
 
-		external, err := cloudprovider.IsCloudProviderExternal(cfg.Infra.Status.PlatformStatus, cfg.FeatureGateAccess)
-		if err != nil {
-			klog.Error(err)
-		} else if external {
-			return "external", nil
-		}
+	if cfg.FeatureGateAccess == nil {
+		panic("FeatureGateAccess is nil")
+	}
 
-		switch cfg.Infra.Status.PlatformStatus.Type {
-		case configv1.AWSPlatformType, configv1.AzurePlatformType, configv1.OpenStackPlatformType, configv1.VSpherePlatformType:
-			return strings.ToLower(string(cfg.Infra.Status.PlatformStatus.Type)), nil
-		case configv1.GCPPlatformType:
-			return "gce", nil
-		default:
+	external, err := cloudprovider.IsCloudProviderExternal(cfg.Infra.Status.PlatformStatus, cfg.FeatureGateAccess)
+	if err != nil {
+		klog.Errorln("failed to check if cloud provider external", err)
+	} else if external {
+		if cfg.CloudControllerDisabled && isCCMNone(cfg.Infra.Status.PlatformStatus) {
```

Comment:
The latter half of this check is already done in [library-go](https://github.com/openshift/library-go/blob/4602d24d27bc94d0cbee0d4646eee867e808c5b4/pkg/cloudprovider/external.go#L45), it's impossible to reach this code on platforms where the cloud controller manager may be disabled.